src/Core/Http/Middleware/AbstractMailMiddleware.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Basement\BetterMails\Core\Http\Middleware;
+
+use Basement\BetterMails\Core\Contracts\BetterMiddlewareContract;
+use Basement\BetterMails\Core\Exceptions\MailException;
+use Closure;
+use Illuminate\Http\Request;
+
+abstract class AbstractMailMiddleware implements BetterMiddlewareContract
+{
+    abstract public function handle(Request $request, Closure $next): mixed;
+
+    /**
+     * @throws MailException
+     */
+    public function validateHeaderKey(array $headers, string $headerKey, string $defaultKey = 'value'): void
+    {
+        $mailUuid = null;
+
+        foreach ($headers as $header) {
+
+            if ($header[$headerKey] == config('filament-better-mails.mails.headers.key')) {
```

Comment:
```suggestion
            if ($header[$headerKey] == strtolower(config('filament-better-mails.mails.headers.key'))) {
```

would be cool to parse the key from the request and ours to lowercase.

> i had a really bad time when i forgot this, so i assume that would be a better thing.